nestkernel/eprop_archiving_node.cpp

```diff
@@ -0,0 +1,253 @@
+  long shift_out = offset_gen_ + delay_in_rec_ + delay_rec_out_;
+
+  return get_name() == "eprop_readout" ? shift_out : shift_rec;
+}
+
+const long
+nest::EpropArchivingNode::get_delay_out_norm() const
+{
+  return delay_out_norm_;
+}
+
+std::map< std::string, double >&
+nest::EpropArchivingNode::get_eprop_parameter_map()
+{
+  return eprop_parameter_map_;
+}
+
+void
+nest::EpropArchivingNode::get_eprop_history( long time_step, std::deque< HistEntryEpropArchive >::iterator* it )
+{
+  if ( eprop_history_.empty() )
+  {
+    *it = eprop_history_.end();
+    return;
+  }
+
+  *it = std::lower_bound( eprop_history_.begin(), eprop_history_.end(), time_step );
+}
+
+double
+nest::EpropArchivingNode::get_firing_rate_reg( long time_step )
+{
+  if ( firing_rate_reg_history_.empty() )
+    return 0.0;
+
+  long const update_interval = kernel().simulation_manager.get_eprop_update_interval().get_steps();
+
+  std::vector< HistEntryEpropFiringRateReg >::iterator it;
+
+  it =
+    std::lower_bound( firing_rate_reg_history_.begin(), firing_rate_reg_history_.end(), time_step + update_interval );
+
+  return it->firing_rate_reg_;
+}
+
+void
+nest::EpropArchivingNode::erase_unneeded_update_history()
+{
+  if ( update_history_.empty() )
+    return;
+
+  for ( auto it = update_history_.begin(); it < update_history_.end(); it++ )
+  {
+    if ( it->access_counter_ == 0 )
+      update_history_.erase( it );
+  }
+}
+
+void
+nest::EpropArchivingNode::erase_unneeded_eprop_history()
+{
+  if ( eprop_history_.empty() or update_history_.empty() )
+    return;
+
+  long update_interval = kernel().simulation_manager.get_eprop_update_interval().get_steps();
+
+  std::deque< HistEntryEpropArchive >::iterator start;
+  std::deque< HistEntryEpropArchive >::iterator finish;
+
+  auto it = update_history_.begin();
+
+  for ( long t = update_history_.begin()->t_; t <= ( update_history_.end() - 1 )->t_; t += update_interval )
+  {
+    if ( it->t_ == t )
+    {
+      it++;
+    }
+    else
+    {
+      get_eprop_history( t, &start );
+      get_eprop_history( t + update_interval, &finish );
+      eprop_history_.erase( start, finish ); // erase found entries since no longer used
+    }
+  }
+  get_eprop_history( 0, &start );
+  get_eprop_history( update_history_.begin()->t_, &finish );
+  eprop_history_.erase( start, finish ); // erase found entries since no longer used
+}
+
+void
+nest::EpropArchivingNode::erase_unneeded_firing_rate_reg_history()
+{
+  if ( update_history_.empty() or firing_rate_reg_history_.empty() )
+    return;
+
+  auto it_update_hist = update_history_.begin();
+  auto it_reg_hist = firing_rate_reg_history_.begin();
+  for ( ; it_update_hist != update_history_.end() and it_reg_hist != firing_rate_reg_history_.end();
+        ++it_update_hist, ++it_reg_hist )
+  {
+    if ( it_update_hist->access_counter_ == 0 )
+      firing_rate_reg_history_.erase( it_reg_hist );
+  }
+}
+
+void
+nest::EpropArchivingNode::add_spike_to_counter()
+{
+  n_spikes_++;
+}
```

Comment:
I'd suggest a clearer name and to use the prefix operator which is slightly simpler than postfix and should be preferred if one does not really need the postfix behaviour.
```suggestion
void
nest::EpropArchivingNode::count_spike()
{
  ++n_spikes_;
}
```